### ISSUE TITLE
Seagull Grounded - Escort

### ISSUE THREAD

**Issue by panfunk** [EdenServer/community#2822]
### Checklist

- [ ] I have searched for existing issues for issues like this one. The issue has not been posted. (Duplicate reports slow down development.)
- [ ] I have provided reproducable steps. (No "as the title says" posts please. Provide reproducable steps even if it seems like a waste of time.)
- [ ] I have provided my client version in the details. (type /ver into your game window)

### Details

We just failed the Assault cuz the NPC ran away, even tho we had one character close on him all the time while the rest went ahead to fight mobs.

from wiki: 
"Only one member of the party needs to escort him. The rest of the party should stay away so they don't interfere with trying to get him moving. Flee or +movement speed gear are useful, but not necessary, for the player staying with the prisoner."

**Comment by KnowOne134** [comment]
this will be tricky until you figure out how to control him, navmesh and npc movement issues, 1 member must stay behind him at a certain distance, if he runs for some reason, mob agro or going out of range you must stop him at his resting points but standing strait in front of him until he says he is caught, thn back up some to make him continue

**Comment by Lixion** [comment]
I think, after pondering it for a bit, does the NPC "lock onto" the first character and that must be the character that controls him? Cause he has right by me in front of him, staring at him, at < 5 yalms or w.e, after he catches his breath. Or do we have to let him get 10 yalms away then reenter the radius to stop him?

**Comment by lindz** [comment]
We had Excaliace stop in the first room and upon seeing mobs, he turned to flee, running in a direct line through the all the walls to the start. By the time I caught up to him, we received the failed message. 

Currently seems impossible to complete if the escort runs through walls, resulting in 40-50y between him and the group within seconds.

**Comment by lindz** [comment]
He also seems to trigger "Now's my chance" each time he stops due to "Someone's been eating too much garlic" despite being at only 5.5y away which will enable him to start moving again.

**Comment by lindz** [comment]
And the newest one, sometimes when he stops due to "Someone's been eating too much garlic" he runs the wrong way once he's triggered to start moving again. He'll run the wrong way a certain distance, then turn and run the correct way again with no dialogue.

**Comment by PoppaPip** [comment]
Casting spells (haste specifically) on Excaliace seems to freeze him in place permanently. I have managed to wrangle him fairly well by always standing at around 3' from him, possibly a sweet spot due to lag?

**Comment by Dorinorin** [comment]
Just tried this assault and Excaliace saw some mobs and ran through the walls to where you couldn’t follow him, then trying to find him results in failure. Not sure if it’s possibly to program the NPC from going through the walls since monsters currently aggro thru floors, walls, and ceilings still.

**Comment by roriffxi** [comment]
He really likes to take off running despite people always being in the proper spot 5-8 yalms away.  Makes thisassault incredibly hard to do.  He'll randomly just start running after standing still and his escort not moving either.  Also at times when he runs, he just bugs out and will not start moving again. just keep repeating the garlic line and does not move.

**Comment by roriffxi** [comment]
Corners are the worst, he loves to bolt when rounding a corner even when someone is right up on him.  This assault is nearly impossible to do reliably because of how badly navmesh or whatever interacts

**Comment by Slamm21** [comment]
tried this today, he definitely seems to be bugged.
He would flee all the time despite me being 5-8 yalms away.

**Comment by rwaffle** [comment]
Having run this about two dozen times, there are 2 buggy things that I can see consistently with him. First Esunii has mentioned in #3365, where you need to stop him at the first door. The second, he will occasionally progress forward without you being able to stop him, and he will ignore monsters, although it isn't very consistent. 
We've also found that depending on your location, latency will add a base of a 2 yalm extension on your distance from him, sometimes up to around 5 yalms. So you need to really be 3-5 yalms from him at all times when moving.

**Comment by falzum** [comment]
Slamm we were able to correct that behavior by keeping our npc person locked on and facing him at all times. 

Granted this was ~6 weeks ago. I can't say if something changed between then and now.

**Comment by Slamm21** [comment]
I did this again, and purposefully stopped him every so often and he never ran away.
seems like maybe he will just run even if you are within the right distance if enough time has passed?

**Comment by E5una** [comment]
Definitely not the case slam, but like Falz said, stay locked on and there shouldn’t be an issue.

In response to what waffle said, yup I’ve seen there before too, I believe it only happens when you stop him and are babysitting him but monsters are in range (aka ur fighting too close). I think if he gets stopped with the “wrong” timing as he detects a monster he will flee to the finish ignoring all monsters, but it may be less strict than that, have rarely seen it. I’ll copy and paste my seagull advice that I’ve DMd people here. It won’t help the devs fix anything but hopefully it’ll help people who end up here and are confused.

(One sec running to my PC then will edit this)

**Comment by Garphilius** [comment]
Yeah, the only real problem (bug) is that first doorway to the right of the start. Of course there are navmesh issues but that is for all zones. You definitely want to stay within 3-5 yalms from him at all times. 6-9 yalms is pushing it and due to latency issues, that 6-9 might actually translate to 9-12 yalms. But yeah, 3-5 yalms and do not worry if you get closer, it is better for him to stop and say the "garlic" text than having him run backwards and wasting more time. 

To avoid the navmesh issues, do not touch any walls and try not to walk on bumps on the ground. 
At the 4-way stops that he does for crab rooms, I just stay 5.1 yalms away (in the main hallway, not on the East or West sides) when he pauses to contemplate which way he is going. If he is going to the right (on the first 4-way), I disengage from lock-on and run right in front of him and he stops. If he is going left (on the first 4-way), I just stay locked-on and continue at a 3-5 yalms distance. Apply this for all 4-ways and you should never have an issue. I also recommend having the rest of the party go up the stairs in each room after killing the crabs and wait until he passes to go investigate the area in that room. As soon as he passes the steps, all PT members head out and go to the next crab room. In this way, all PT members will never be near him so you do not have to worry about having other members in range and possibly screwing up with his 10 yalm radius tractor as the escort person will always be the only one near him.

**Comment by Slamm21** [comment]
This has gotten way worse lately, for example, today he ran for no reason (we were within 2 yalms) much after the door, he then did the pant message but instead of the damn message, he kept running.
After the second time he did this, we had 2 people on top of him and he did it again and exited.
This happens probably one in every 5 runs for me right now.

**Comment by Slamm21** [comment]
A few times I have seen the npc ignore 1 character, so you have to bring another to it to stop it.
I'm wondering if my above comment was because of this, and also the server tick of 3 seconds. When I arrived at the npc (the other person was always next to it) it was about 2 seconds before he ran away, maybe the server tick happened just before I got in range?

**Comment by Slamm21** [comment]
![image](https://user-images.githubusercontent.com/5552081/104953216-de684b00-59bd-11eb-9d3e-41204ce0156b.png)
heres one today (he ran again after that pant message when winrie stayed right next to him like that the whole time.

After we cleared the 3rd crab room we ran to winrie who had stopped him in the centre bit just before he chooses which room for march and ballads. He then did the "Now's My Chance" as soon as we ran towards the fish, even though winrie was still standing on top of him, and despite following him constantly he just kept going until the exit.

I wonder if whichever character triggers the garlic message is the one that has to stay with him? so if you run past the npc it might trigger the garlic message and when you get far enough the npc could run away even though someone else is on top of the npc maybe. Might explain why sometimes the npc cannot be stopped unless a 2nd person runs back to catch it.

**Comment by Slamm21** [comment]
LoS changes seem to have fixed this, happy days

**Comment by Midgard-was-taken** [comment]
NPC behaviour still seems like a complete crapshoot. My man just doesn't want to be in there and GTFOs no matter what if he's not feeling it that day.

**Comment by KnowOne134** [comment]
He doesn't want to be there, I feel his pain.

Your escorting him to prison and there is this icky dark cave full of monsters that want to kill all over you. He sees his chance and starts running for doors like Scooby-Doo.

**Comment by Mozzy4Ever** [comment]
Gunna reference #4339 cause it has a video of what's happening. Isn't right, but it's still possible to complete the assault. Have done it multiple times for people who say it's not :P

**Comment by randomslumber** [comment]
1. Don't cast anything on him or his AI will break and he'll do nothing indefinitely
2. Stay right up on his ass ~4yalms facing him especially around corners or other areas where your sixth sense tells you the navmash might say you don't have LOS on him. Otherwise he will try to flee and waste time. He's probably like monsters where he's actually ahead of where he is on your screen
